src/EnergyPlus/BoilerSteam.cc

```diff
@@ -98,11 +98,11 @@ namespace BoilerSteam {
         }
 
         // Now look for this particular pipe in the list
-        for (auto &boiler : state.dataBoilerSteam->Boiler) {
-            if (boiler.Name == objectName) {
-                return &boiler;
-            }
-        }
+        auto myBoiler = std::find_if(state.dataBoilerSteam->Boiler.begin(),
+                                     state.dataBoilerSteam->Boiler.end(),
+                                     [&objectName](const BoilerSpecs &boiler) { return boiler.Name == objectName; });
+        if (myBoiler->Name == objectName) return myBoiler;
```

Comment:
You’re doing the string comparison twice, once in find_if and once here. A better test is if (myBoiler != state.dataBoilers->Boilers.end()).